src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -37,33 +37,40 @@ Copyright (c) .NET Foundation. All rights reserved.
     ============================================================
     -->
 
-  <PropertyGroup Condition="'$(NoBuild)' != 'true'">
-    <_BeforePublishTargets>Build</_BeforePublishTargets>
-  </PropertyGroup>
-
-  <PropertyGroup Condition="'$(NoBuild)' == 'true'">
+  <PropertyGroup>
     <!-- We still need to resolve references even if we are not building during publish. -->
     <!-- BuildOnlySettings are required for RAR to find satellites and dependencies -->
-    <_BeforePublishTargets>
+    <_BeforePublishNoBuildTargets>
       BuildOnlySettings;
       _PreventProjectReferencesFromBuilding;
       ResolveReferences;
       PrepareResourceNames;
       ComputeIntermediateSatelliteAssemblies;
-    </_BeforePublishTargets>
+    </_BeforePublishNoBuildTargets>
+
+    <_CorePublishTargets>
+      PrepareForPublish;
+      ComputeAndCopyFilesToPublishDirectory;
+      GeneratePublishDependencyFile;
+      GeneratePublishRuntimeConfigurationFile;
+    </_CorePublishTargets>
   </PropertyGroup>
 
+  <Target Name="PublishBuild"
```

Comment:
These should be _ ("private") and still thinking about the names.

Reply:
`_PublishBuildAlternative _PublishNoBuildAlternative` ?? So people look at the following line will know only one of them will be run. And by appending `alternative`, it looks pretty internal

```xml
<Target Name="Publish"
   Condition="$(IsPublishable) == 'true'"
  DependsOnTargets="$(_PublishBuildAlternative);$(_PublishNoBuildAlternative)" 
```

Reply:
I like those names :)